exonum-java-binding/testkit/src/test/java/com/exonum/binding/testkit/TestKitTest.java

```diff
@@ -345,6 +348,31 @@ void createBlockWithTransactions() {
     }
   }
 
+  @Test
+  void findTransactionsInPool() {
+    try (TestKit testKit = TestKit.forService(TestServiceModule.class)) {
+      TestService service = testKit.getService(TestService.SERVICE_ID, TestService.class);
+
+      String messagePayload = "Test message";
+      TransactionMessage message = constructTestTransactionMessage(messagePayload);
+      RawTransaction rawTransaction = TestKit.toRawTransaction(message);
+      TransactionMessage message2 = constructTestTransactionMessage("Test message 2");
+      RawTransaction rawTransaction2 = TestKit.toRawTransaction(message2);
+
+      try {
+        service.getNode().submitTransaction(rawTransaction);
+        service.getNode().submitTransaction(rawTransaction2);
+      } catch (InternalServerError e) {
+        fail(e);
+      }
+
+      List<TransactionMessage> transactionsInPool =
+          testKit.findTransactionsInPool(
+              tx -> Arrays.equals(tx.getPayload(), messagePayload.getBytes(BODY_CHARSET)));
```

Comment:
Also, if during migration of service ITs we will determine that this method is useful, we might add some `TransactionPreconditions`: `hasSamePayloadAs(TransactionMessage)`, `hasSamePayloadAs(RawTransaction)`, `hasServiceId(int)`, etc.

However, we must also consider what is better (e.g., in terms of code size, or error reporting) — let users use 
```
assertThat(testkit.getPool()).
    <Some library matcher, possibly, with our predicates>
```
or `findTransactionsInPool` :thinking:

Reply:
Yes, this method proved useful, although almost always I wanted to just get all in-pool transactions of a certain service. But users might have other cases - so the current method seems good as it is for now.